test/Microsoft.DotNet.ShellShim.Tests/ShellShimMakerTests.cs

```diff
@@ -38,15 +39,16 @@ public void GivenAnRunnerOrEntryPointItCanCreateConfig(string entryPointPath, st
             if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                 return;
 
-            var shellShimMaker = new ShellShimMaker(TempRoot.Root);
+            var cleanFolderUnderTempRoot = GetNewCleanFolderUnderTempRoot();
+            var shellShimMaker = new ShellShimMaker(cleanFolderUnderTempRoot);
 
-            var tmpFile = Path.Combine(TempRoot.Root, Path.GetRandomFileName());
+            var tmpFile = new FilePath(Path.Combine(cleanFolderUnderTempRoot, Path.GetRandomFileName()));
```

Comment:
All these small lines are due to I need a cleanFolderUnderTempRoot, so I can asset "there is nothing left on disk if install failed"